README.md

```diff
@@ -6,41 +6,184 @@
 
-To enable logging in `DEBUG` level, add selected categories to list in `Help` -> `Diagnostic Tools` -> `Debug Log Settings`. <br/>
-A relatively small amount of `TRACE`-level logs is generated as well.
+![](docs/machete_file_editor.gif)
 
+For more information about `machete` file look at the [reference blog post](https://medium.com/virtuslab/make-your-way-through-the-git-rebase-jungle-with-git-machete-e2ed4dbacd02)
 
-## Development
+### Checking out branches
+
+With Git Machete you can easily check out branches that have been defined in `machete` file.
+Just right-click on the branch you wont to check out and select `Checkout Branch`.
+You can also double-click on the selected branch to check it out.
+Current branch is underlined in branch tree.
+
+![](docs/checkout.gif)
+
+### Listing commits
+
+Git Machete can display commits belonging to the branches.
+It displays only these commits that are unique for the selected branch (i.e. were never a tip of a different branch).
+When you click on eye button on the left side of plugin, you can toggle between show and hide commits.
+
+![](docs/toggle_listing_commits.gif)
+
+
+### Rebase
+
+Rebase with this plugin is easy!
+Just right-click on the child branch and from context menu select `(Checkout and) Rebase Branch onto Parent` and a standard IntelliJ dialog for interactive rebase will appear.
+
+![](docs/rebase.gif)
+
+### Push
+
+After a rebase, you might want to push rebased branch to remote.
+To do this, right-click on the branch you want to push and select `Push (Current) Branch` from context menu.
+Standard IntelliJ push dialog will appear but in case when we rebased selected branch (when branch we want to push diverge from its remote) only force push button will be available.
+
+![](docs/push.gif)
+
+### Pull
+
+If branch is behind (or in sync with) its remote, it can be pulled.
+To do this, right-click on given branch and select `Pull Branch`.
+
+![](docs/pull.gif)
+
+### Reset to remote
+
+If given branch is not in sync with its remote, we can reset it to point to the exact same commit as the remote.
+From right-click context menu select `Checkout and Reset Branch to Remote`.
+This action performs `git reset --keep` under the hood. `--keep`, as opposed to `--hard`, keeps your uncommitted changes safe from getting overwritten.
+
+![](docs/reset.gif)
+
+### Fast forward parent
+
+When you are done with the changes on the selected branch (e.g. PR was approved and all required fixes are applied), you probably want to merge this branch to its parent branch.
+When these branches are in sync (green edge) you can do a fast-forward merge.
+This is what `Fast Forward Parent Branch To Match Current Branch` context menu action do.
+As you can see, edge between given branch and its parent was changed to grey - it means that the child branch has been merged.
+
+![](docs/fast_forward.gif)
+
+### Slide out branch
```

Comment:
What about documenting the `machete.slideOut.deleteLocalBranch`?

Reply:
@PawelLipski, should it be official feature?

Reply:
Yeah, let's make it so